### PR TITLE
Make clippy happy

### PR DESCRIPTION
**Pull request by 0xPoe**
Please check each commit.

### CODE REVIEW

File: src/currentprocess/filesource.rs [hunk 1]
```diff
@@ -24,8 +24,8 @@ impl Stdin for io::Stdin {
     fn lock(&self) -> Box<dyn StdinLock + '_> {
         Box::new(io::Stdin::lock(self))
     }
-    fn read_line(&self, buf: &mut String) -> Result<usize> {
-        io::Stdin::read_line(self, buf)
+    fn read_line(&self, buf: &mut str) -> Result<usize> {
+        io::Stdin::read_line(self, &mut buf.to_owned())
```
Comment (kinnison) [inline src/currentprocess/filesource.rs]: This is confusing, surely that won't work?

File: src/currentprocess/filesource.rs [hunk 2]
```diff
@@ -68,8 +68,8 @@ impl Stdin for TestStdin {
             inner: self.0.lock().unwrap_or_else(|e| e.into_inner()),
         })
     }
-    fn read_line(&self, buf: &mut String) -> Result<usize> {
-        self.lock().read_line(buf)
+    fn read_line(&self, buf: &mut str) -> Result<usize> {
+        self.lock().read_line(&mut buf.to_owned())
```
Comment (kinnison) [inline src/currentprocess/filesource.rs]: Ditto, won't this end up losing the read-in content?

File: src/diskio/immediate.rs [hunk 3]
```diff
@@ -204,7 +204,7 @@ impl IncrementalFileWriter {
     fn write(&mut self, chunk: Vec<u8>) -> std::result::Result<bool, io::Error> {
         let mut state = self.state.lock().unwrap();
         if let Some(ref mut state) = *state {
-            if let Some(ref mut file) = (&mut self.file).as_mut() {
+            if let Some(ref mut file) = (self.file).as_mut() {
```
Comment (kinnison) [inline src/diskio/immediate.rs]:
```suggestion
            if let Some(ref mut file) = self.file.as_mut() {
```
?

File: src/dist/signatures.rs [hunk 4]
```diff
@@ -56,20 +56,12 @@ impl VerificationHelper for Helper<'_> {
...
-                            _ => {
-                                // We ignore any errors.
-                            }
-                        }
+                    for GoodChecksum { ka, .. } in results.into_iter().flatten() {
+                        // A good signature!  Find the index
+                        // of the singer key and return
+                        // success.
+                        self.index = self.certs.iter().position(|c| c.cert() == ka.cert());
+                        assert!(self.index.is_some());
```
Comment (kinnison) [inline src/dist/signatures.rs]: This loses the nice explicit indicator that we are ignoring errors during signature verification, which is an open question for us.  We shouldn't hide that.  If you're going to do this refactor then please ensure there's a comment to the effect that the `.flatten()` causes us to ignore errors for now.  Also fix the comment `singer` -> `signer` please :D

**Review by kinnison — CHANGES_REQUESTED** [review]

File: src/currentprocess/filesource.rs (hunk 1 above)
Comment (0xPoe) [reply src/currentprocess/filesource.rs]: Yes, this should be a false positive.

**Review by kinnison — APPROVED** [review]

**Pull request merged**